File: Easy/LC1351/src/LC1351.java
/*
 * LC #1351 Probem Link - https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
 * 
 * @author ketanmalik
 * May 26 2020
 * 
 * */

public class LC1351 {
	public static void main(String[] args) {
		System.out.println(countNegatives(new int[][] { { 3, 1 }, { 1, -1 } }));
	}

	public static int countNegatives(int[][] grid) {
		int row = grid.length, col = grid[0].length, res = 0;
		for (int i = 0; i < row; i++) {
			for (int j = 0; j < col; j++) {
				if (grid[i][j] < 0) {
					res += col - j;
					break;
				}
			}
		}
		return res;
	}

}

File: Easy/LC1351/LC1351.js
/*
 * LC #1351 Probem Link - https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
 *
 * @author ketanmalik
 * May 26 2020
 *
 * */

var countNegatives = function (grid) {
  let row = grid.length,
    col = grid[0].length,
    res = 0;
  let i = row - 1,
    j = 0;
  while (i >= 0 && j < col) {
    let num = grid[i][j];
    if (num >= 0) j++;
    else {
      i--;
      res += col - j;
      j = 0;
    }
  }
  return res;
};

console.log(
  countNegatives([
    [5, 1, 0],
    [-5, -5, -5],
  ])
);
